distributed/deploy/tests/test_adaptive.py

```diff
@@ -460,3 +424,58 @@ async def test_update_adaptive(cleanup):
         await asyncio.sleep(0.2)
         assert first.periodic_callback is None
         assert second.periodic_callback.is_running()
+
+
+from distributed.utils_test import slowinc
+
+
+@pytest.mark.asyncio
+async def test_scale_needs_to_be_awaited(cleanup):
+    """
+    This tests that the adaptive class works fine if the scale method uses the
+    `sync` method to schedule its task instead of loop.add_callback
+    """
+
+    class RequiresAwaitCluster(LocalCluster):
+        def scale(self, n):
+            # super invocation in the nested function scope is messy
+            method = super().scale
+
+            async def _():
+                return method(n)
+
+            return self.sync(_)
+
+    async with RequiresAwaitCluster(n_workers=0, asynchronous=True) as cluster:
+        async with Client(cluster, asynchronous=True) as client:
+            futures = client.map(slowinc, range(5), delay=0.05)
+            assert len(cluster.workers) == 0
+            cluster.adapt()
+
+            await client.gather(futures)
+
+            del futures
+            await async_wait_for(lambda: not cluster.workers, 10)
+
+
+@pytest.mark.asyncio
+async def test_adaptive_stopped():
```

Comment:
Given that this uses an async cluster, do we need the `async_wait_for` + timeout or can we just `assert` the various attributes directly? For example, when I make the following changes locally this test still passes:

```diff
diff --git a/distributed/deploy/tests/test_adaptive.py b/distributed/deploy/tests/test_adaptive.py
index 39284302..8cf4f381 100644
--- a/distributed/deploy/tests/test_adaptive.py
+++ b/distributed/deploy/tests/test_adaptive.py
@@ -466,16 +466,8 @@ async def test_adaptive_stopped():
     """
     async with LocalCluster(n_workers=0, asynchronous=True) as cluster:
         async with Client(cluster, asynchronous=True) as client:
-            instance = cluster.adapt(interval="10ms")
+            pc = cluster.adapt(interval="10ms").periodic_callback
+            assert pc is not None
+            assert pc.is_running() is not None

-            await async_wait_for(
-                lambda: instance.periodic_callback is not None, timeout=5
-            )
-
-            await async_wait_for(
-                lambda: instance.periodic_callback.is_running() is not None, timeout=5
-            )
-
-            pc = instance.periodic_callback
-
-    await async_wait_for(lambda: pc.is_running() is not None, timeout=5)
+    assert pc.is_running() is not None
```

Reply:
Well, at the very least we'll need to have one wait in between which actually waits for the entire thing to start. If the PC was never started, the conditions are trivially true. I'll add a comment and what I can remove for it to still work

Reply:
All of the `is not None` checks where useless which is why the test did not fail. I properly assert for bool now and now I need to wait